### PR TITLE
chore: update docker to v26

### PR DESCRIPTION
**Pull request by matejvasek**
* Updated docker dep v25 => v26
* Removed usage of deprecated `github.com/docker/distribution/reference` package.

### CODE REVIEW

**Comment by matejvasek** [comment]
cc @coreydaley

**Comment by matejvasek** [comment]
cc @adambkaplan

File: go.mod [hunk 1]
```diff
@@ -4,8 +4,8 @@ go 1.20
 
 require (
 	github.com/containers/image/v5 v5.29.0
-	github.com/docker/distribution v2.8.3+incompatible
-	github.com/docker/docker v25.0.3+incompatible
+	github.com/distribution/reference v0.5.0
```
Comment (adambkaplan) [inline go.mod:7]: :+1: to docker/distribution moving to the CNCF! :tada:

**Review by adambkaplan — APPROVED** [review]
/lgtm

**Pull request merged**